### DISCUSSION TITLE
Frequency-dependent equivalent linear analysis

### DISCUSSION THREAD

**Discussion by jf-huang** [Caltech-geoquake/PySeismoSoil#11] [General]
Hi, does PySeismoSoil support the _frequency-dependent_ equivalent linear analysis or has any plan to do so?

**Comment by jsh9** [comment]
Hi @jf-huang , it does not currently support frequency-dependent equivalent linear.  I have the frequency-dependent equivalent linear code in MATLAB, but I'm not sure I have the time to translate that into Python.

@domniki and @yzshi98: do you have plans or interests in potentially working on this?  (Or does anyone else in the group has interests?)  Thanks!

**Comment by yzshi98** [comment]
Hi Jian, I may not work on the translation, because my research direction is more about AI based ground motion synthesis. The other PhD student "Flora Xia" in Domniki Group, who works on the SeismoVlab and structural dynamics has more experience in frequency-dependent equivalent linear method. But I am not sure whether she has the time to finish the translation, you can email Domniki for confirmation.

**Comment by domniki** [comment]
Hi Everybody and apologies, Jian, for not getting back to you sooner in response to this subject,

I am working on a proposal to improve on the EQ FD method in part using ML to better characterize the smooth variation of the high frequency decay function. I think we can do much better than 2000 version of our algorithm and once we have it in place, we will have a student implement it on pySeismoSoil. The python translation is not an issue.

I just don’t think it’s worth doing so before we implement the improvements I have in mind.

Thanks for your continuing interest and support of our work!
Domniki

Domniki Asimaki (she/her/hers)
Professor of Mechanical and Civil Engineering
California Institute of Technology
Gates-Thomas Room 355
1200 E. California Boulevard<https://www.google.com/maps/search/1200+E.+California+Boulevard?entry=gmail&source=g>, MC 104-44
Pasadena, CA 91125
phone: (626) 395-2742
zoom: https://caltech.zoom.us/my/domniki
website: http://asimaki.caltech.edu
http://orcid.org/0000-0002-3008-8088


From: Sycamore ***@***.***>
Date: Monday, June 19, 2023 at 10:13 AM
To: jsh9/PySeismoSoil ***@***.***>
Cc: Asimaki, Domniki (Domniki) ***@***.***>, Mention ***@***.***>
Subject: Re: [jsh9/PySeismoSoil] Frequency-dependent equivalent linear analysis (Discussion #11)

Hi Jian, I may not work on the translation, because my research direction is more about AI based ground motion synthesis. The other PhD student "Flora Xia" in Domniki Group, who works on the SeismoVlab and structural dynamics has more experience in frequency-dependent equivalent linear method. But I am not show whether she has the time to finish the translation, you can email Domniki for confirmation.

**Comment by jf-huang** [comment]
Hi everyone, I appreciate your quick response and also looking forward to the new algorithm. Thanks again!
